### ISSUE TITLE
Can't update to field to null

### ISSUE THREAD

**Issue by sandrosimas** [boblauer/mongoose-dateonly#6]
This is my code: 
````
exports.updatePaymentOrder = async (paymentOrder, data) => {
  console.log(data)
  const attrs = _.pick(data, ALLOWED_ATTRS);
  const loadedPaymentOrder = await paymentOrderQueries.getPaymentOrder(paymentOrder.id);
  await loadedPaymentOrder.update(attrs, {runValidators: true});
  loadedPaymentOrder.set(attrs);
  return loadedPaymentOrder;
};
````
This is the log when the function above is executed:
````
2018-07-21T23:15:37.190Z - debug: PUT /5b4fa13c0ec5e844757333d5 [Account: 5b368d9458b52c13789eddda]
Mongoose: accounts.findOne({ _id: ObjectId("5b368d9458b52c13789eddda") }, { fields: { company: 1 } })
Mongoose: payment_orders.findOne({ _id: ObjectId("5b4fa13c0ec5e844757333d5") }, { fields: { company: 1 } })
{ payment_date: null }
Mongoose: payment_orders.findOne({ _id: ObjectId("5b4fa13c0ec5e844757333d5") }, { fields: {} })
Mongoose: payment_orders.update({ _id: ObjectId("5b4fa13c0ec5e844757333d5") }, { '$set': { payment_date: 20180621 } }, { runValidators: true })
2018-07-21T23:15:37.226Z - debug: PUT /payment_orders/5b4fa13c0ec5e844757333d5 200 236 - 67.041 ms
````
Look at the output of ````console.log````:
````
{ payment_date: null }
````
Mongoose is updating the field ````payment_date```` to 20180621. Why this is happening?
````
Mongoose: payment_orders.update({ _id: ObjectId("5b4fa13c0ec5e844757333d5") }, { '$set': { payment_date: 20180621 } }, { runValidators: true })
````
This is my model:
````
const PaymentOrder = new Schema({
  company: {type: ObjectId, ref: 'Company', required: true},
  client: {type: ObjectId, ref: 'Client', required: true, index: true},
  method: {type: String, required: true},
  amount: {type: Number, required: true},
  due_date: {type: DateOnly},
  payment_date: {type: DateOnly},
  registration_date: {type: Date, required: true},
}, {collection: 'payment_orders'});
````

**Comment by sandrosimas** [comment]
I think this code from the library should check if value exists:
https://github.com/boblauer/mongoose-dateonly/blob/d1f6168d77ce285771eeec345dc11b0ced58e6d4/lib/schema.js#L18-L24

**Comment by sandrosimas** [comment]
And maybe here:
https://github.com/boblauer/mongoose-dateonly/blob/d1f6168d77ce285771eeec345dc11b0ced58e6d4/lib/schema.js#L40-L51

**Comment by boblauer** [comment]
It does indeed look like that is the source of the issue. I'm fairly busy these days so I can't make any promises on when I'll be able to get to this issue, but a pull request would certainly be welcome!

**Comment by sandrosimas** [comment]
@boblauer, I will send a PR until tomorrow, OK?

**Comment by boblauer** [comment]
That would be great, thank you!

**Comment by sandrosimas** [comment]
@boblauer, I sent a pull request #7

**Comment by boblauer** [comment]
Thanks for the pull! Fix published to npm, `@1.0.5`.
